### ISSUE TITLE
Disabled ComboBox allows keyboard focus

### ISSUE THREAD

**Issue by v-atkulk** [microsoft/fluentui#13685]
“[Check out Accessibility Insights!](https://accessibilityinsights.io/) - Identify accessibility bugs before check-in and make bug fixing faster and easier.”

**GitHubTags:**#A11yTCS;#A11yMAS;#A11ySev2;#Benchmark#DesktopWeb;#MAS2.1.1;#Keyboard;#FTP;#STP;#Chrome;#ChromiumEdge;Win10;

Environment Details:​
Application Name: Notebooks - Azure machine learning service
Windows Version: Windows10​
Browser version: Microsoft Edge  Version 85.0.545.0 (Official build) dev (64-bit))
Chrome Version Version 83.0.4103.106 (Official Build) (64-bit)

**Repro Steps:​**
1.Launch the workspace URL
2.Navigate to Compute in the left panel under manage and click on new button.
3.Create a new compute.
4.Click on home in the left  panel and then on start now button under notebook.
5.Select  .ipynb file under user files.
6.Check if the keyboard focus is shifted to disabled controls in the menu bar.

**Actual:**
When we navigate through the menu bar using tab keys or arrow keys the disabled controls also receive keyboard focus.

Same issue observed in following pages:
Notebook->Subset of header.
1. Launch   the workspace URL
2. aaaaaaa.txt file will get opened.
3. Verify that all the elements of "Subset of header" are accessible and MAS Compliant.

**Same issue observed in Chrome also.**

**Expected:**
The disabled controls should not receive keyboard focus when navigated using tab or arrow keys.

**User Impact:**
Keyboard as well as screen reader users will get confused as the disabled controls receive keyboard focus.

**Attachment for Reference:**
[13685_A11y_Notebooks - Azure machine learning service_Notebook_Menu bar_Keyboard.zip](https://github.com/microsoft/fluentui/files/4800144/13685_A11y_Notebooks.-.Azure.machine.learning.service_Notebook_Menu.bar_Keyboard.zip)

[13685_A11y_Notebooks - Azure machine learning service_Notebook_Menu bar_ScreenReader.zip](https://github.com/microsoft/fluentui/files/4800150/13685_A11y_Notebooks.-.Azure.machine.learning.service_Notebook_Menu.bar_ScreenReader.zip)

**Comment by khmakoto** [comment]
Thanks for filing this, but you'll need to add a CodePen repro before we can help. You can start from one of the component examples in our [website](https://developer.microsoft.com/en-us/fluentui#/controls/web/) (click "Export to CodePen") or https://aka.ms/fluentpen.

**Comment by v-atkulk** [comment]
Here is the CodePen repro for the bug: https://codepen.io/Douglas2020/pen/PoZpBZX 
You could also just use screen reader like NVDA, and tab to the disabled Combo-box, will find that is still readable.

**Comment by DouglasXiao** [comment]
Hi @khmakoto , please kindly refer to the above CodePen link and here is few of my analysis.

**Root cause**:
The Fabric UI component ComboBox does not support attribute like "allowDisabledFocus" as other components.

**Suggested Solution**:
Inside ComboBox, Autofill component is rendered, and inside Autofill component, is using <input .../>. If can put tabindex={-1} inside the <input .../> based on the input props like "allowDisabledFocus", the disabled ComboBox won't be tab-able then.

**Comment by JustSlone** [comment]
@v-atkulk and @DouglasXiao 
As I understand it, the correct accessibility behavior is to focus the disabled control. Otherwise users using screen readers would not know the item exists at all. This is especially true for buttons, but I believe it also extends to ComboBoxes. 

Other components (such as Button) didn't use to behave correctly, so the "allowDisabledFocus" was added to enable this accessible behavior in a non-breaking way. 

Can you elaborate on the use case for skipping the disabled control?

**Comment by v-atkulk** [comment]
Hi @JustSlone 
All the controls present on the screen receive screen reader focus whether they are in disabled state or in focusable state.
If the control is disabled it can be accessed using (caps+arrow) in narrator and arrow keys in NVDA.
Here the control is navigable using tab key it means that the control is still enabled/ focusable.
Request you to check the focusable property for the button/combo-box so that the focus won't go to the control if it is disabled.

**Comment by JustSlone** [comment]
Got it, checked with folks knowledgeable on the Accessibility behavior here and we've confirmed that a disabled ComboBox should in-fact not be focusable via Tab. 

The fix of adding tabIndex="-1" on the input element when the ComboBox is disabled also seems reasonable. 

@v-atkulk or @DouglasXiao would you be interested in submitting a PR for this fix?

**Comment by DouglasXiao** [comment]
Hi @JustSlone , would be glad to do that, would you mind sharing the convention and more details of how to contribute to this repository?

**Comment by JustSlone** [comment]
Yeah, check out our wiki for setup instructions and contributing guidelines https://github.com/microsoft/fluentui/wiki/Setup#contributing-changes it should have what you need to know. 
You may also need to sign the Microsoft Contributor License Agreement info here: https://github.com/microsoft/fluentui/wiki/CLA it's pretty easy. 

If you have any questions or if the wiki is unclear let me know and I'll happily help (and correct the wiki too).

**Comment by vivianysun** [comment]
Hello! 
I'm also seeing this same behavior for buttons in the command bar where command bar menu items are tab focusable when menu items are disabled. Will this get fixed with this bug here, or should I file a new issue? Thanks!

**Comment by JustSlone** [comment]
@vivianysun That will be a separate fix and should be a separate issue. Thanks!

**Comment by ecraig12345** [comment]
Seems like a duplicate of #13137.
